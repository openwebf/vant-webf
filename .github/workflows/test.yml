name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
