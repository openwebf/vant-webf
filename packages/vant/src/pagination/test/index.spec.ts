import { mount } from '../../../test';
import { Pagination } from '..';

test('should render prev-text、next-text slot correctly', () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 50,
      showPageSize: 5,
    },
    slots: {
      'prev-text': () => 'Custom PrevText',
      'next-text': () => 'Custom NextText',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render page slot correctly', () => {
  const wrapper = mount(Pagination, {
    props: {
      totalItems: 50,
      showPageSize: 5,
    },
    slots: {
      page: ({ text }) => `foo ${text}`,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should emit change event after the page is changed', async () => {
  const wrapper = mount(Pagination, {
    props: {
      modelValue: 1,
      totalItems: 50,
      'onUpdate:modelValue': (modelValue: number) => {
        wrapper.setProps({ modelValue });
      },
    },
  });

  await wrapper
    .findAll('.van-pagination__item--page button')[2]
    .trigger('click');
  expect(wrapper.emitted('change')).toEqual([[3]]);

  await wrapper.find('.van-pagination__item--prev button').trigger('click');
  expect(wrapper.emitted('change')).toEqual([[3], [2]]);

  await wrapper.find('.van-pagination__item--next button').trigger('click');
  expect(wrapper.emitted('change')).toEqual([[3], [2], [3]]);
});
