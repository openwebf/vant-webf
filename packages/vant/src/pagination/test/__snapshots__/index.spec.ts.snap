// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page slot correctly 1`] = `
<nav role="navigation"
     class="van-pagination"
>
  <ul class="van-pagination__items">
    <li class="van-pagination__item van-pagination__item--prev van-hairline--surround">
      <button type="button">
        Previous
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        foo 1
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        foo 2
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        foo 3
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        foo 4
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        foo 5
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--next van-hairline--surround">
      <button type="button">
        Next
      </button>
    </li>
  </ul>
</nav>
`;

exports[`should render prev-text„ÄÅnext-text slot correctly 1`] = `
<nav role="navigation"
     class="van-pagination"
>
  <ul class="van-pagination__items">
    <li class="van-pagination__item van-pagination__item--prev van-hairline--surround">
      <button type="button">
        Custom PrevText
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        1
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        2
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        3
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        4
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--page van-hairline--surround">
      <button type="button">
        5
      </button>
    </li>
    <li class="van-pagination__item van-pagination__item--next van-hairline--surround">
      <button type="button">
        Custom NextText
      </button>
    </li>
  </ul>
</nav>
`;
